src/main/drivers/system.c

```diff
@@ -60,16 +60,60 @@ void cycleCounterInit(void)
     usTicks = clocks.SYSCLK_Frequency / 1000000;
 }
 
+#include "common/atomic.h"
+#include "debug.h"
+
+static volatile int sysTickRollover = 0;
+
 // SysTick
 void SysTick_Handler(void)
 {
-    sysTickUptime++;
+    ATOMIC_BLOCK(NVIC_PRIO_MAX) {
+        sysTickUptime++;
+        sysTickRollover = 0;
+        *(volatile uint32_t *)&(SysTick->CTRL);
```

Comment:
`SysTick->CTRL` is already declared `volatile`.
You may cast it to `(void)` to document the read ..